### ISSUE TITLE
Asterisk (AMI) pjsip mode for centreon

### ISSUE THREAD

**Issue by ermurenz** [centreon/centreon-plugins#4672]
Hi, i don't know if this is the correct place for this and if is not, i apologies in advanced.
And I apologize for my low level English too.
We have numerous Asterisk servers with version 16 with “sip” configured.
Centreon plugin “Asterisk VoIP Server” with mode “sip-peers-usage” , “channel-usage” and "dahdi-status" is perfect for us.
(This https://docs.centreon.com/pp/integrations/plugin-packs/procedures/applications-voip-asterisk-ami/)

This is the command line example that we use:

/usr/lib/centreon/plugins/centreon_asterisk_ami.pl --plugin=apps::voip::asterisk::ami::plugin  --mode=**sip-peers-usage** --ami-hostname=xxxxxxxxxxxxx --ami-username=xxxxx  --ami-password=xxxxx  --warning-status='%{status} =~ /LAGGED|UNKNOWN/i' --critical-status='%{status} =~ /UNREACHABLE/i' 

Using Asterisk Manager Interface (AMI) we can monitor all problems we can have, both with virtual servers and physical ones with dadhi.
The problem is that we are migrating to a more recent verison of Asterisk (20.4.0) and we are moving to the pjsip protocol but it seems that centreon doesn’t have a proper mode for this.
In other words, If we use the same “mode” mentioned before, we get this error:

“UNKNOWN: Communication issue [Message: Command output follows -- No such command 'sip show peers' (type 'core show help sip show' for other possible commands)]”

And this make sense because Asterisk, for the pjsip protocol, uses a different command like:

*CLI> pjsip show endpoints
*CLI> pjsip xxxx xxxx

and so on.
For what i can see,  there’s no “ready” check/mode for this but maybe I'm wrong.
Maybe there is or it's in development and I'm the one who doesn't know where to find it.
I know one thing: we feel desperate without in-depth monitoring.

Thanks in advance.
Emanuele.

**Comment by garnier-quentin** [comment]
If you have commands for each modes and the output results, we could manage it

**Comment by ermurenz** [comment]
ok Thank you, i appreciate.
I'm going to prepare the information and then I'll post it as soon as I have it.

**Comment by ermurenz** [comment]
If someone knows more informations about pjsip commands and the various 'status' , you are welcome.
Anyway, these are most important ones that we use:



1 - "pjsip show registrations"

 <Registration/ServerURI..............................>  <Auth....................>  <Status.......>
==========================================================================================

 REGNAME/sip:voip.fqdn.xx:5060                          REGNAME-AUTH                  Registered        (exp. 3131s)

Objects found: 1

2 - "pjsip show endpoints"

 Endpoint:  <Endpoint/CID.....................................>  <State.....>  <Channels.>
    I/OAuth:  <AuthId/UserName...........................................................>
        Aor:  <Aor............................................>  <MaxContact>
      Contact:  <Aor/ContactUri..........................> <Hash....> <Status> <RTT(ms)..>
  Transport:  <TransportId........>  <Type>  <cos>  <tos>  <BindAddress..................>
   Identify:  <Identify/Endpoint.........................................................>
        Match:  <criteria.........................>
    Channel:  <ChannelId......................................>  <State.....>  <Time.....>
        Exten: <DialedExten...........>  CLCID: <ConnectedLineCID.......>
==========================================================================================

 Endpoint:  ENDPOINTNAME01                                               Not in use    0 of inf
        Aor:  ENDPOINTNAME01                                             1
      Contact:  ENDPOINTNAME01/sip:ENDPOINTNAME01@IPADDRESS:5060       ef3dcfdcfa NonQual         nan
  Transport:  transport-udp             udp      0      0  0.0.0.0:5060
   Identify:  ENDPOINTNAME01/ENDPOINTNAME01
        Match: IPADDRESS/32

 Endpoint:  ENDPOINTNAME02                                               Not in use    0 of inf
        Aor:  ENDPOINTNAME02                                             1
      Contact:  ENDPOINTNAME02/sip:ENDPOINTNAME02@IPADDRESS:5060       c659a89750 NonQual         nan
  Transport:  transport-udp             udp      0      0  0.0.0.0:5060
   Identify:  ENDPOINTNAME02/ENDPOINTNAME02
        Match: IPADDRESS/32


Objects found: 2

3 - "pjsip show contacts"

  Contact:  <Aor/ContactUri..............................> <Hash....> <Status> <RTT(ms)..>
==========================================================================================

  Contact:  name/sip:name@IPADDRESS:5060          6e1beb0f3f NonQual         nan
  Contact:  name/sip:000000000@voip.eutelia.it           c9c6337a38 NonQual         nan

Objects found: 2


4 - "pjsip show aors"

      Aor:  <Aor..............................................>  <MaxContact>
    Contact:  <Aor/ContactUri............................> <Hash....> <Status> <RTT(ms)..>
==========================================================================================

      Aor:  NAME 01                                               1
    Contact:  NAME01/sip:NAME01@IPADDRESS:5060        6e1beb0f3f NonQual         nan

      Aor:  NAME                                                1
    Contact:  NAME/sip:00000000@voip.FQDN.it         c9c6337a38 NonQual         nan


Objects found: 2



The formatting of the output part isn't the best but it was copied and pasted as it was.
Hope it's enough but if you need further information don't hesitate to ask.

**Comment by ermurenz** [comment]
Hi,
I know that it takes time and it's not something that can be done in a few hours, also because ultimately we rely on your availability and your work commitments, but I was wondering if by any chance there was hope that some developer is keeping an eye on it,
Anyway, even if this isn't the case, I thank you for your support and patience.

**Comment by omercier** [comment]
Hi,
Thanks for your interest in Centreon.
I have noticed that you have already created an idea on theWatch, so I'm closing this issue and we'll see how this gets prioritized.
Thank you for your understanding.
